src/utils/src/files_functions.rs

```diff
@@ -0,0 +1,242 @@
+use std::{path::{PathBuf, Path}, io::Cursor, ffi::OsStr};
+
+use dioxus::desktop::DesktopContext;
+use futures::StreamExt;
+use mime::*;
+use tokio_util::io::ReaderStream;
+use warp::{constellation::{Progression, directory::Directory}, error::Error};
+use image::io::Reader as ImageReader;
+
+use crate::Storage;
+
+pub async fn upload_file(file_storage: Storage, file_path: PathBuf, eval_script: DesktopContext) {
```

Comment:
Instead of cloning, you could do the following

```suggestion
pub async fn upload_file(mut file_storage: Storage, file_path: PathBuf, eval_script: DesktopContext) {
```

Reply:
Done!